app/components/Form/ProjectContactForm.tsx

```diff
@@ -301,6 +291,39 @@ const ProjectContactForm: React.FC<Props> = (props) => {
     );
   }, [projectRevision.projectContactFormChanges]);
 
+  const deleteContact = (
+    formChangeId: string,
+    formChangeOperation: FormChangeOperation
+  ) => {
+    discardFormChange({
+      formChange: { id: formChangeId, operation: formChangeOperation },
+      onCompleted: () => {
+        delete formRefs.current[formChangeId];
+        // Will remove the primary contact form change if it is not filled out and we have only one secondary contact form
+        // This is to fix the issue where the primary contact form change is not removed when the secondary contact is deleted
+        if (
+          !primaryContactForm.newFormData.contactId &&
+          alternateContactForms.length === 1
+        ) {
+          undoFormChanges({
+            variables: {
+              input: {
+                formChangesIds: formChangeIds,
+              },
+            },
+            onCompleted: () => {
```

Comment:
We've had issues in the past with nested `onCompleted` callbacks (mostly because the closure is not up-to-date with the form data once the first onComplete call resolves - as you noticed `alternateContactForms.length === 1` should actually be 0 at that point)

We should be able to decide beforehand:

```tsx
  const deleteContact = (
    formChangeId: string,
    formChangeOperation: FormChangeOperation
  ) => {
    if(primaryContactForm.newFormData.contactId ||  alternateContactForms.length > 1)
      discardFormChange(...)
    else
      undoFormChanges(...)
  }
```

Reply:
Absolutely!

Reply:
also, it'll be much easier to test 😁